app/components/FeatureDetailHeader.js

```diff
@@ -63,6 +63,65 @@ export default class FeatureDetailHeader extends React.Component {
     navigation.navigate('Explore', { feature, mode })
   }
 
+  renderEditGeometryButton () {
+    const { feature, navigation } = this.props
+    const { pendingDeleteUpload } = feature.properties
+
+    // Hide edit geometry button on AddFeatureDetail screen
+    if (navigation.state.routeName === 'AddFeatureDetail') {
+      return
+    }
+
+    // If feature was deleted and waiting to be uploaded, block further edits
+    // by passing button without associated action.
+    if (pendingDeleteUpload) {
+      return <Button>
+        <Coordinates>
+          <Edit>This feature was deleted</Edit>
+        </Coordinates>
+      </Button>
+    }
+
+    // Get feature type to allow/disallow editing
+    let featureType
+    switch (feature.geometry.type) {
+      case 'Point':
+        featureType = 'node'
+        break
+      case 'LineString':
+        featureType = 'way'
+        break
+      case 'Polygon':
+        featureType = 'way'
+        break
+      default:
+        featureType = 'unsupported'
+        break
+    }
+
+    if (featureType === 'unsupported') {
+      // If feature is unsupported, block further edits
+      return <Button>
+        <Coordinates>
+          <Edit>This feature was deleted</Edit>
```

Comment:
@vgeorge should this instead say something like `Editing this feature is not supported`?

Reply:
Of course. I updated it now.